lib/compress.js

```diff
@@ -2592,9 +2592,10 @@ merge(Compressor.prototype, {
         }
         // x && (y && z)  ==>  x && y && z
         // x || (y || z)  ==>  x || y || z
-        if (self.right instanceof AST_Binary
-            && self.right.operator == self.operator
-            && (self.operator == "&&" || self.operator == "||"))
+        if (self instanceof AST_Binary
+         && self.right instanceof AST_Binary
+         && self.right.operator === self.operator
+         && (self.operator === "&&" || self.operator === "||"))
```

Comment:
So you prefer to indent instead of alignment by those `&&`?

Just asking as I don't mind either way.